### PR TITLE
Add Notifications to OBALocationManager

### LINKED ISSUE #676: Remove OBALocationManager's multiple delegates system
`OBALocationManager` uses an unusual-for-iOS pattern: multiple delegates. This system has some issues. For instance, its delegates are all strongly referenced instead of being weakly referenced.

I'd love to see OBA move to a more normal-for-iOS system, like using `NSNotification`s for this sort of thing. Proper resolution of this issue will include the following:

- [ ] Remove `OBALocationManager`'s add and remove methods for delegates, and associated data structures, and any other obsolete code
- [ ] Definition of a new notification name
- [ ] Replacement of the delegate calls in `OBALocationManager` with messages to `NSNotificationCenter`
- [ ] Ideal/Possibly Optional: A small number of test cases that demonstrate that the new notification system works properly.

### PR DESCRIPTION
**Pull request by aaronbrethorst**
Fixes #676

Remove the multiple delegates pattern from OBALocationManager and replace it with notifications.

### CODE REVIEW

File: OBAKit/Location/OBARegionHelper.m [hunk 1]
```diff
@@ -17,15 +17,28 @@ @interface OBARegionHelper ()
...
+
+- (void)dealloc {
+    [self unregisterFromLocationNotifications];
+}
+
 - (void)updateNearestRegion {
     [self updateRegion];
-    OBALocationManager *lm = [OBAApplication sharedApplication].locationManager;
-    [lm addDelegate:self];
+    OBALocationManager *lm = self.locationManager;
```
Comment (chadsy) [inline OBAKit/Location/OBARegionHelper.m]: Maybe just `[self.locationManager startUpdatingLocation];` ?

File: OBAKit/Location/OBALocationManager.m [hunk 2]
```diff
@@ -123,28 +108,37 @@ - (void)handleNewLocation:(CLLocation*)location {
...
+        [self broadcastDidUpdateLocation:_currentLocation];
+    }
 }
 
-@end
+#pragma mark - Notifications
+
+- (void)broadcastDidUpdateLocation:(CLLocation*)location {
+    [[NSNotificationCenter defaultCenter] postNotificationName:OBALocationDidUpdateNotification object:self userInfo:nil];
+}
```
Comment (chadsy) [inline OBAKit/Location/OBALocationManager.m]: Is there any precedent for these being posted on the main thread?

File: OBAKit/Location/OBARegionHelper.h [hunk 3]
```diff
@@ -17,8 +19,15 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)regionHelperShowRegionListController:(OBARegionHelper*)regionHelper;
 @end
 
-@interface OBARegionHelper : NSObject <OBALocationManagerDelegate>
+@interface OBARegionHelper : NSObject
+@property(nonatomic,strong) OBALocationManager *locationManager;
+@property(nonatomic,strong) OBAModelDAO *modelDAO;
+@property(nonatomic,strong) OBAModelService *modelService;
```
Comment (chadsy) [inline OBAKit/Location/OBARegionHelper.h]: maybe `readonly` on these 3?

File: OBAKit/Location/OBARegionHelper.m (hunk 1 above)
Comment (aaronbrethorst) [inline OBAKit/Location/OBARegionHelper.m]: 👍

**Pull request merged**